### PR TITLE
Update Semgrep CI with custom Semgrep rules dirs/repos

### PR DESCRIPTION
**Pull request by ahpaleus**
Closes: #32

### CODE REVIEW

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 1]
```diff
@@ -35,6 +35,60 @@ The following are key points to note:
...
+passed to the `SEMGREP_RULES`, follow the steps below:
+
+1. If your custom Semgrep rules directory **is in the same** repository as the scanned code,
+       just pass the directory path in the `SEMGREP_RULES` variable:
+      (e.g., `SEMGREP RULES: p/default custom-semgrep-rules-dir/`)
+
+2. If your custom Semgrep rules **are in another private repository**, do the following:
+
+    a. [Generate an access token for the repository with Semgrep rules](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+    Remember to select the least scopes necessary.
```
Comment (elopez) [inline content/docs/static-analysis/semgrep/20-ci.md:52]: Maybe we should be explicit as to what those scopes are.
```suggestion
    Remember to select the least scopes necessary (e.g. a fine-grained token for your repository with read-only access over the repository contents).
```

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 2]
```diff
@@ -35,6 +35,60 @@ The following are key points to note:
...
+        with:
+          repository: ${{ github.repository_owner }}/${{ env.SEMGREP_PRIVATE_RULES_REPO }} # organization-name/Semgrep-rules-repo
+          token: ${{ secrets.SEMGREP_RULES_TOKEN }} # Configured PAT
+          path: ${{ env.SEMGREP_PRIVATE_RULES_REPO }} # Relative path to place the repository
+        # ...
+      - run: semgrep ci
+      env:
+        # Pass the directory with the checked-out Semgrep rules repository
+        SEMGREP_RULES: ${{ env.SEMGREP_PRIVATE_RULES_REPO }}
+    ```
```
Comment (elopez) [inline content/docs/static-analysis/semgrep/20-ci.md:91]: The indentation is incorrect on the `env` section here

```suggestion
      - run: semgrep ci
        env:
          # Pass the directory with the checked-out Semgrep rules repository
          SEMGREP_RULES: ${{ env.SEMGREP_PRIVATE_RULES_REPO }}
    ```
```

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 3]
```diff
@@ -35,6 +35,60 @@ The following are key points to note:
...
+      SEMGREP_PRIVATE_RULES_REPO: semgrep-private-rules
+    steps:
+        # Main checkout of the repository source code
+      - name: Checkout main repository
+        uses: actions/checkout@v4
+        # Checkout of the repository with custom Semgrep rules
+      - name: Checkout private custom Semgrep rules
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.repository_owner }}/${{ env.SEMGREP_PRIVATE_RULES_REPO }} # organization-name/Semgrep-rules-repo
```
Comment (elopez) [inline content/docs/static-analysis/semgrep/20-ci.md:83]: example did not match the env variable

```suggestion
          repository: ${{ github.repository_owner }}/${{ env.SEMGREP_PRIVATE_RULES_REPO }} # organization-name/semgrep-private-rules
```

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 4]
```diff
@@ -78,14 +132,32 @@ jobs:
...
-          SEMGREP_RULES: p/default # Browse more rulesets - semgrep.dev/explore
+          # Set the SEMGREP_RULES environment variable to specify which rules Semgrep should use.
+          # Use common security-related rulesets for this job (starting with `p/`)
+          # or use a directory with your custom rules from the current repository (such as `semgrep-rules/`).
+          SEMGREP_RULES: >
+            p/default
+          # If you have a directory in the current repo with your custom rules:
+          # semgrep-rules/
+        # Pass the directory with the checked-out Semgrep rules repository
+        #    ${{ env.SEMGREP_PRIVATE_RULES_REPO }}
```
Comment (elopez) [inline content/docs/static-analysis/semgrep/20-ci.md:160]: indentation

```suggestion
          # If you have a directory in the current repo with your custom rules:
          # semgrep-rules/
          # Pass the directory with the checked-out Semgrep rules repository
          #  ${{ env.SEMGREP_PRIVATE_RULES_REPO }}
```

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 5]
```diff
@@ -35,6 +35,60 @@ The following are key points to note:
...
+       just pass the directory path in the `SEMGREP_RULES` variable:
+      (e.g., `SEMGREP RULES: p/default custom-semgrep-rules-dir/`)
+
+2. If your custom Semgrep rules **are in another private repository**, do the following:
+
+    a. [Generate an access token for the repository with Semgrep rules](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+    Remember to select the least scopes necessary.
+
+    b. [Add the generated access token](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions)
+    to the repository where the workflow is run
```
Comment (elopez) [inline content/docs/static-analysis/semgrep/20-ci.md:55]:
```suggestion
    as a secret to the repository where the workflow is run.
```

**Review by elopez — COMMENTED** [review]
I have just a few comments, but this looks good overall 👍

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 6]
```diff
@@ -35,6 +35,60 @@ The following are key points to note:
...
+
+    {{< hint info >}}
+If your repository with custom rules is publicly available, just omit the steps where you create the PAT and do not pass
+the `token` in the checkout step.
+    {{< /hint >}}
+
+    For example:
+
+    ```yaml
+    # Set up an environment variable with the name of the private repository with custom Semgrep rules
```
Comment (fcasal) [inline content/docs/static-analysis/semgrep/20-ci.md:72]: with -> containing

File: content/docs/static-analysis/semgrep/20-ci.md [hunk 7]
```diff
@@ -35,6 +35,60 @@ The following are key points to note:
...
+    env:
+      SEMGREP_PRIVATE_RULES_REPO: semgrep-private-rules
```
Comment (fcasal) [inline content/docs/static-analysis/semgrep/20-ci.md:74]: Does it need to be under the same org? Or can you do other-user/semgrep-private-rules?

Comment (fcasal) [reply content/docs/static-analysis/semgrep/20-ci.md:74]: Nevermind, I see below

**Comment by fcasal** [comment]
lgtm after @elopez 's comments are addressed

File: content/docs/static-analysis/semgrep/20-ci.md (hunk 6 above)
Comment (ahpaleus) [reply content/docs/static-analysis/semgrep/20-ci.md]: https://github.com/trailofbits/testing-handbook/pull/34/commits/827891b7fd39aac737b779af752222320da4bf8a

**Pull request merged**